pkg/minikube/machine/cache_binaries.go

```diff
@@ -63,7 +64,7 @@ func KubernetesReleaseURLSHA1(binaryName, version, osName, archName string) stri
 
 // CacheBinary will cache a binary on the host
 func CacheBinary(binary, version, osName, archName string) (string, error) {
-	targetDir := localpath.MakeMiniPath("cache", version)
+	targetDir := localpath.MakeMiniPath(filepath.Join("cache", osName), version)
```

Comment:
I think you can use `MakeMiniPath("cache", osName, version)` here

Reply:
thanks, updated!